docs/DeveloperGuide.adoc

```diff
@@ -74,42 +70,191 @@ The sections below give more details of each component.
+
+==== Log UI View
+
+.Structure of the Log UI View
+image::UiLogClassDiagram.png[]
+
+The Log View is made up of the link:{repoURL}/blob/master/src/main/java/seedu/address/ui/panel/log/LogPanel.java[`LogPanel`] which contains several instances of `LogBox`.
+
+link:{repoURL}/blob/master/src/main/java/seedu/address/ui/panel/log/LogBox.java[`LogBox`] displays literal information back to the user when it is called by `MainWindow` when it listens for a command.
 
 [[Design-Logic]]
-=== Logic component
+=== CommandManager component
 
-[[fig-LogicClassDiagram]]
-.Structure of the Logic Component
-image::LogicClassDiagram.png[]
+==== Command
+
+A `Command` is defined to be an immutable function, that can be invoked at any time, to perform any set of instructions. After executing the set of instructions, it will optionally return output to be displayed to the user.
+
+In Horo, a `Command` implemented as an abstract class with an abstract `execute()` method. To create a new concrete command, extend from `Command` and implement `execute()`.
+
+.Class diagram of Commands
+image::CommandManager.png[]
+
+If your concrete command requires any dependencies during execution, it is recommended to pass in the dependency from the constructor.
+
+For example, if your command needs to be able to access `ModelManager`:
+```
+public class MyCommand extends Command {
+	private final ModelManager model;
+	MyCommand(ModelManager model) {
+		this.model = model;
+	}
+
+	@Override
+    CommandOutput execute() {
+        this.model.doSomething();
+		...
+	}
+}
+```
+
+
+==== CommandParser
+
+A `CommandParser` is defined to be able to parse a `String` of user input, and return a `Command`.
+
+In Horo, a `CommandParser` is implemented as a finite state machine (FSM). It parses user input token by token, and it transitions from state to state depending on the next token.
+
+What the FSM is trying to do is tokenize user input into:
+
+1. `Command keyword`
+2. `Command phrase(s)`
+
+A *command keyword* is defined as the first sequence of consecutive, non-whitespace characters of the user input. For the rest of this guide, a sequence of consecutive, non-whitespace characters will be referred to as a ‘word’. A word can be described in the form of a regular expression (regex) as `[^\s]+`.
+
+Examples of valid command keywords:
+
+- `exit`
+- `add_event`
+- `123`
+- `😺`
+
+A *command phrase* is defined as either a word, or multiple words delimited by whitespace surrounded by quotes. Command phrases come after a command keyword.
+
+Examples of valid command phrases:
+
+- `Horse`
+- `”Homework”`
+- `”Horo’s Birthday”`
+- `”24/10/2019 07:00”`
+- `’24/10/2019 07:00’`
+- `--description`
+
+A valid command contains exactly one command keyword, and zero or more command phrases.
+
+`[keyword] [phrase] [phrase] [phrase] ...`
+
+.State diagram of CommandParser
+image::CommandParser.png[]
```

Comment:
Perhaps you would like write the state in each node (the rounded rectangles) so that it is clearer